fdbserver/SkipList.cpp

```diff
@@ -40,10 +34,8 @@
 #include "fdbclient/SystemData.h"
 #include "fdbserver/Knobs.h"
 
-#define PARALLEL_THREAD_COUNT 0	// FIXME: When >1, program execution (e.g. random numbers) is/was nondeterministic.  Why?
```

Comment:
I personally think the comment is useful knowledge for future developers, such as warning what may go wrong if they want to tweak the code.

Reply:
I'm confused by this comment. I removed all of the code that reads this macro.

Reply:
I just don't know if we will ever want to explore `PARALLEL_THREAD_COUNT  > 0`. I **assume** when `PARALLEL_THREAD_COUNT  > 0`, we can have multiple threads working on the skip list.  The comment means the skip list is not thread safe.

Another question I have for other people who may know the history of `PARALLEL_THREAD_COUNT `: What is the motivation of exploring `PARALLEL_THREAD_COUNT  > 0`? Have we concluded that `PARALLEL_THREAD_COUNT  > 0` won't work?

Reply:
The `PARALLEL_THREAD_COUNT` code has been dead for as long as the code is in GitHub and increases the code complexity for no value. See for example:

https://github.com/apple/foundationdb/pull/2630/files/f5be0db366d24a6eb7e6cc82a6de096f97f98db0..3f6cacb477cfc58883be5ae5758222cfef7a2baa#diff-4bb1e76b464c10a924b2d2c1cdd739b5L878
https://github.com/apple/foundationdb/pull/2630/files/f5be0db366d24a6eb7e6cc82a6de096f97f98db0..3f6cacb477cfc58883be5ae5758222cfef7a2baa#diff-4bb1e76b464c10a924b2d2c1cdd739b5L836

In order to revive that code, someone will need to refer back to this PR where the comment will still be present.